### ISSUE TITLE
Is SynonymousTo deprecated or removed?

### ISSUE THREAD

**Issue by mmnosek** [Starcounter/Home#360]
I have quite an old project, working on Starcounter-RC-2.3.0.6157. I tried to run it on 2.3.1.8329 (current stable), but I found an issue related to `SynonymousTo` depreciation. Shouldn't it work as before, but with some warning? It also relies on `Simplified`. Some errors:
* `A synonym targeted a non-persistent field. Test.Card.Code, synonymous to Simplified.Ring1.Something.Name (ScErrSynTargetNotPersistent (SCERR4070))`
* `A synonym targeted a non-persistent field. Starcounter.Authorization.Database.PermissionSomebodyGroup.Permission, synonymous to Simplified.Ring1.Relation.WhatIs (ScErrSynTargetNotPersistent (SCERR4070)`

These messages are not explaining the problem.

**Comment by Mackiovello** [comment]
The [docs](https://docs.starcounter.io/v/2.4/cookbook/how-to-migrate-from-synonymousto) say that "The [SynonymousTo] attribute was deprecated in Starcounter 2.4". If this happened in 2.3.1, the docs are wrong.

**Comment by miyconst** [comment]
The docs are wrong. Here is the state of `[SynonymousTo]`:

- Deprecated on paper in `2.3.1`.
- Deprecated in code in `2.3.2` (going to).
- Removed in code in `2.4` (going to).

**Comment by mmnosek** [comment]
If it's deprecated in `2.3.1`, why I have such errors?

**Comment by miyconst** [comment]
Can you point to the source code, please?

**Comment by miyconst** [comment]
I just have checked and the outcome is the following:

- Database fields are still supported.
- `SynonymousTo` over a database field is supported, only if the targeted column is also a field.
- If the parent column is declared as a property, then `SynonymousTo` fails with `A synonym targeted a non-persistent field `.
- Applying `SynonymousTo` over a property fails with `Illegal element for the SynonymousTo attribute`.

In short: **stop using `SynonymousTo` and all these exceptions won't happen**.

In long: `SynonymousTo` never supported database properties, only fields. Database fields are still supported. So, everything is backwards compatible.

This all applies to `2.3.1`.

Closing the issue as answered.
